Framework/src/runPostProcessingOCC.cxx

```diff
@@ -0,0 +1,216 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    runPostProcessingOCC.cxx
+/// \author  Piotr Konopka
+///
+/// \brief This is an OCC executable to run postprocessing
+
+#include "QualityControl/PostProcessingRunner.h"
+#include "QualityControl/QcInfoLogger.h"
+
+#include <OccInstance.h>
+#include <RuntimeControlledObject.h>
+#include <boost/program_options.hpp>
+#include <Common/Timer.h>
+
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::postprocessing;
+using namespace AliceO2::Common;
+namespace bpo = boost::program_options;
+
+class PostProcessingOCCStateMachine : public RuntimeControlledObject
+{
+ public:
+  PostProcessingOCCStateMachine(std::string name, std::string configPath, double period = 1.0)
+    : RuntimeControlledObject("Post-processing task runner"), mName(name), mConfigPath(configPath), mPeriod(period)
+  {
+    mRunner = std::make_unique<PostProcessingRunner>(name, configPath);
+  }
+
+  // In all of the methods below we handle exceptions, so we can go into error state - OCC won't do that for us.
```

Comment:
Maybe this is something we could discuss at some point with Teo because it makes the code a bit less fancy than it could.

Reply:
Yes, I have to think if there is any case when exceptions should cause transition to ERROR state, now I can't think of any.